metaflow/plugins/kfp/interruptible_decorator.py

```diff
@@ -1,4 +1,18 @@
+import os
+from typing import Optional
+
+import requests
 from metaflow.decorators import StepDecorator
+from metaflow.metadata.metadata import MetaDatum
+
+
+def _get_ec2_metadata(path: str) -> Optional[str]:
+    response = requests.get(f"http://169.254.169.254/latest/meta-data/{path}")
```

Comment:
What is `http://169.254.169.254`. Should this be parameterized?

Reply:
Ah I see... It seems to be a special IP address for EC2 instance's metadata. Could you please add this info as a comment in code? 

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html

Reply:
> The 169.254.169.254 IP address is a “magic” IP in the cloud world, in [AWS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html) it used to retrieve user data and instance metadata specific to a instance. It can only be accessed locally from instances and available without encryption and authentication.

* https://avatao.com/blog-security-issues-to-be-aware-of-before-moving-to-the-cloud/#:~:text=The%20169.254.,available%20without%20encryption%20and%20authentication.
* https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html